internal/registry/registry.go

```diff
@@ -548,6 +651,23 @@ func (r *Registry) ProviderInfo(providerID peer.ID) (*ProviderInfo, bool) {
 	return pinfo, r.policy.Allowed(providerID)
 }
 
+// ExtendedProviders returns Extendedproviders record associated with the peerID.
+func (r *Registry) ExtendedProviders(providerID peer.ID) *ExtendedProviders {
+	infoChan := make(chan *ExtendedProviders)
+	r.actions <- func() {
+		extendedProviders, ok := r.extendedProviders[providerID]
+		if ok {
+			infoChan <- extendedProviders
+		}
+		close(infoChan)
+	}
+	extendedProviders := <-infoChan
+	if extendedProviders == nil {
+		return nil
+	}
+	return extendedProviders
+}
+
```

Comment:
There may be a couple of functions below there need modification:
- I think the `ImportProviders` function needs to be able to import extended providers as well.
- The `RemoveProvider`function should also remove all the associated extended providers.

These concerns avove would automatically be handled if `ExtendedPRoviders` was part of the `ProviderInfo` type

Reply:
Agreed, that'd be a better solution. Updated the code.